ipn/ipnlocal/local_test.go

```diff
@@ -820,3 +821,29 @@ type legacyBackend interface {
 // Verify that LocalBackend still implements the legacyBackend interface
 // for now, at least until the macOS and iOS clients move off of it.
 var _ legacyBackend = (*LocalBackend)(nil)
+
+func TestWatchNotificationsCallbacks(t *testing.T) {
+	b := new(LocalBackend)
+	n := new(ipn.Notify)
+	b.WatchNotifications(context.Background(), 0, func() {
+		// Ensure a watcher has been installed.
+		if len(b.notifyWatchers) != 1 {
```

Comment:
this is a potential data race I think? This callback is called with the lock not held, so you should probably grab it here. 

And everything else in this test: line 835, and 846?